mobile_rewards/src/server.rs

```diff
@@ -267,48 +247,98 @@ impl Server {
-                }
-            }
+            .source_unordered(5, manifests)
+            .filter_map(|msg| async move {
+                msg.map_err(|err| {
+                    tracing::error!("Error fetching reward manifest: {:?}", err);
+                    err
+                })
+                .ok()
+            })
+            .filter_map(|msg| async move {
+                RewardManifest::decode(msg).map_or_else(
+                    |err| {
+                        tracing::error!("Could not decode reward manifest: {:?}", err);
+                        None
+                    },
+                    Some,
+                )
+            })
+            .filter(move |manifest| {
+                future::ready(manifest.start_timestamp >= (last_reward_manifest as u64))
+            })
+            .flat_map_unordered(5, |manifest| {
+                stream::iter(
+                    manifest
+                        .written_files
+                        .into_iter()
+                        .map(|file_name| FileInfo::from_str(&file_name)),
+                )
+            })
+            .boxed();
+        let reward_shares = self
+            .verifier_store
+            .source_unordered(5, reward_files)
+            .filter_map(|msg| async move {
+                msg.map_err(|err| {
+                    tracing::error!("Error fetching reward share: {:?}", err);
+                    err
+                })
+                .ok()
+            })
+            .filter_map(|msg| async move {
+                RadioRewardShare::decode(msg).map_or_else(
+                    |err| {
+                        tracing::error!("Could not decode reward share: {:?}", err);
+                        None
+                    },
+                    Some,
+                )
+            });
+
+        tokio::pin!(reward_shares);
+
+        let mut owner_rewards: BTreeMap<(u64, u64), HashMap<Vec<u8>, u64>> = BTreeMap::new();
+
+        while let Some(reward_share) = reward_shares.next().await {
+            *owner_rewards
+                .entry((reward_share.start_epoch, reward_share.end_epoch))
+                .or_default()
+                .entry(reward_share.owner_key)
+                .or_default() += reward_share.amount;
         }
 
-        if !rewards.is_empty() {
-            self.issue_rewards(rewards, reward_period).await?;
-            last_reward_end_time
-                .update(&self.pool, time_range.end.timestamp())
-                .await?;
-        } else {
-            tracing::error!("cannot continue; unable to determine reward period!");
-            return Err(Error::NotFound("invalid reward period".to_string()));
+        for (epoch, rewards) in owner_rewards.into_iter() {
+            let rewards = rewards
+                .into_iter()
+                .map(|(account, amount)| SubnetworkReward { account, amount })
+                .collect();
+            self.issue_rewards(rewards, epoch.0..epoch.1).await?;
```

Comment:
this is where you still need to pass in the "reward_period" range of block heights to construct the txn. this range is retrieved at the top of `process_clock_tick` by querying the follower which gets it from the chain

Reply:
I literally keep forgetting this every single time